app/controllers/articles_controller.rb

```diff
@@ -1,5 +1,9 @@
 class ArticlesController < ApplicationController
 
+  http_basic_authenticate_with name: "dhh",
+                               password: "secret",
+                               except: [:index, :show]
```

Comment:
Not the most sophisticated example of authentication but kudos to the tutorial for at least addressing the topic.